src/m365/commands/login.ts

```diff
@@ -243,7 +243,13 @@ class LoginCommand extends Command {
       }
     }
     finally {
-      if (!args.options.ensure || (args.options.ensure && !auth.service.connected)) {
+      if (!args.options.ensure ||
+        (args.options.ensure && !auth.service.connected) ||
+        (args.options.ensure && (args.options.userName && args.options.userName !== auth.service.userName)) ||
+        (args.options.ensure && (args.options.certificateFile && (auth.service.certificate !== fs.readFileSync(args.options.certificateFile as string, 'base64')))) ||
+        (args.options.ensure && (args.options.appId && args.options.tenant && (args.options.appId !== auth.service.appId) && (args.options.tenant !== auth.service.tenant))) ||
```

Comment:
Won't this check prevent the ability to use a different multitenant app without specifying the `tenantId`?